### ISSUE TITLE
0.6.3: Pytest fails while testing if HelioviewerClient is online - even when running tests offline.

### ISSUE THREAD

**Issue by olebole** [sunpy/sunpy#1726]
When running the CI tests, `sunpy/net/tests/test_helioviewer.py` fails while trying to determine whether the client is online (`client.is_online()`); the web page seems to return an HTML page insted of an JSON object. This can be reproduced with
```
from sunpy.net.helioviewer import HelioviewerClient
client = HelioviewerClient()
client.is_online()
```
This raises the exception
```
/usr/lib/python2.7/dist-packages/sunpy/net/helioviewer.pyc in is_online(self)
    218         """Returns True if Helioviewer is online and available."""
    219         try:
--> 220             self.get_data_sources()
    221         except urllib2.URLError:
    222             return False

/usr/lib/python2.7/dist-packages/sunpy/net/helioviewer.pyc in get_data_sources(self, **kwargs)
     32         params.update(kwargs)
     33 
---> 34         return self._get_json(params)
     35 
     36     def get_closest_image(self, date, **kwargs):

/usr/lib/python2.7/dist-packages/sunpy/net/helioviewer.pyc in _get_json(self, params)
    227         """Returns a JSON result as a string"""
    228         response = self._request(params).read()
--> 229         return json.loads(response)
    230 
    231     def _get_file(self, params, directory=None, overwrite=False):

/usr/lib/python2.7/dist-packages/sunpy/net/helioviewer.pyc in _get_json(self, params)
    227         """Returns a JSON result as a string"""
    228         response = self._request(params).read()
--> 229         return json.loads(response)
    230 
    231     def _get_file(self, params, directory=None, overwrite=False):

/usr/lib/python2.7/json/__init__.pyc in loads(s, encoding, cls, object_hook, parse_float, parse_int, parse_constant, object_pairs_hook, **kw)
    337             parse_int is None and parse_float is None and
    338             parse_constant is None and object_pairs_hook is None and not kw):
--> 339         return _default_decoder.decode(s)
    340     if cls is None:
    341         cls = JSONDecoder

/usr/lib/python2.7/json/decoder.pyc in decode(self, s, _w)
    362 
    363         """
--> 364         obj, end = self.raw_decode(s, idx=_w(s, 0).end())
    365         end = _w(s, end).end()
    366         if end != len(s):

/usr/lib/python2.7/json/decoder.pyc in raw_decode(self, s, idx)
    380             obj, end = self.scan_once(s, idx)
    381         except StopIteration:
--> 382             raise ValueError("No JSON object could be decoded")
    383         return obj, end

ValueError: No JSON object could be decoded
```

Also, in the test `client.is_online()` is called even when the `not online` flag is given on the command line.

**Comment by Cadair** [comment]
I think this is a duplicate of #1724 the API has moved out from underneath us.

**Comment by Cadair** [comment]
Reopening for the mis-marked as offline test.

**Comment by bsipocz** [comment]
The 3-liner works for me on both py3.4 and 2.7 on the branch of #1731.

**Comment by Cadair** [comment]
yeah, there is a fix in master and in 0.6 now. The problem is that this should not be running if the tests are in offline mode.
